SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,906 @@
++    pTracking->nPrevElement = 2;
++
++    if (!pTracking->nInPackage || pTracking->nPrintPackage)
++    {
++        // print end element to file
++        printElementEndToFile(pTracking->pbOutfile, name);
++
++    } 
++    else if (pTracking->nInPackage && !pTracking->nTimeFound)
++    {
++        int nEndElementBufferLen = 500;
++        pszElementBuffer = malloc(sizeof(char) * nEndElementBufferLen);
++        if (!pszElementBuffer)
++        {
++            dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++            BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++        }
++        // add end element to buffer
++        int nEndElementLen = addElementEndToBuffer(pszElementBuffer, &nEndElementBufferLen, name);
++
++        checkAndResizeBuffer(&(pTracking->pszElementBuffer), &(pTracking->nBufferMaxLen), nEndElementLen);
++        strcat(pTracking->pszElementBuffer, pszElementBuffer);
++
++    } // else do nothing
++
++    if (strcmp(name, "package") == 0)
++    { // on end package, reset tracking function
++        // reset userData
++        pTracking->nBufferLen = 0;
++        bzero(pTracking->pszElementBuffer, pTracking->nBufferMaxLen);
++        pTracking->nInPackage = 0;
++        pTracking->nPrintPackage = 0;
++        pTracking->nTimeFound = 0;
++    }
++cleanup:
++    if (pszElementBuffer)
++    {
++        free(pszElementBuffer);
++    }
++    return;
++error:
++    goto cleanup;
++}
++
++void TDNFFilterCharDataHandler(void * userData, const char * content, int length) {
++    uint32_t dwError = 0;
++    // load tracking data
++    TDNFFilterData * pTracking = (TDNFFilterData *)userData;
++    pTracking->nPrevElement = 1;
++
++    char * pszCharData = malloc((length + 1) * sizeof(char));
++    if (!pszCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    bzero(pszCharData, (length + 1) * sizeof(char));
++    strncpy(pszCharData, content, length);
++    char * pszLintedCharData = descEscapeCharLinter(pszCharData);
++    if (!pszLintedCharData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    // check params 
++    if (!pTracking->nInPackage || pTracking->nPrintPackage)
++    {
++        // print to file
++        fprintf(pTracking->pbOutfile, "%s", pszLintedCharData);
++    }
++    else if (pTracking->nInPackage && !pTracking->nTimeFound)
++    {
++        // add to buffer
++        checkAndResizeBuffer(&(pTracking->pszElementBuffer), &(pTracking->nBufferMaxLen), length);
++        strcat(pTracking->pszElementBuffer, pszLintedCharData);
++    } // else do nothing (skipped package)
++
++cleanup:
++    if (pszLintedCharData)
++    {
++        free(pszLintedCharData);
++    }
++    if (pszCharData)
++    {
++        free(pszCharData);
++    }
++    return;
++error:
++    goto cleanup;
++}
++
++char * 
++TDNFFilterFile(const char * pszInFilePath, char * pszEpochTime) {
```

Comment:
const char * on the pszEpochTime argument

Reply:
added